metrics/histogram.go

```diff
@@ -0,0 +1,112 @@
+package wavefront
+
+import (
+	"log"
+	"sync"
+	"time"
+
+	"github.com/wavefronthq/wavefront-sdk-go/senders"
+
+	tdigest "github.com/caio/go-tdigest"
+)
+
+type Histogram struct {
+	mutex              sync.Mutex
+	priorTimedBinsList []*timedBin
+	currentTimedBin    *timedBin
+
+	metricName string
+	source     string
+	tags       map[string]string
+
+	Granularity senders.HistogramGranularity
+	Compression uint32
+	MaxBins     int
+}
+
+type timedBin struct {
+	tdigest   *tdigest.TDigest
+	timestamp time.Time
+}
+
+type distribution struct {
+	centroids []senders.Centroid
+	timestamp time.Time
+}
+
+func NewHistogram(metricName string, source string, tags map[string]string) *Histogram {
+	h := &Histogram{}
+	h.MaxBins = 10
+	h.Granularity = senders.MINUTE
+	h.Compression = 5
+
+	h.metricName = metricName
+	h.source = source
+	h.tags = tags
+
+	return h
+}
+
+func (h *Histogram) Add(v float64) {
+	h.rotateCurrentTDigestIfNeedIt()
+	h.currentTimedBin.tdigest.Add(v)
+}
+
+func (h *Histogram) Report(sender senders.Sender) {
+	distributions := h.getDistributions()
+	log.Printf("metric: '%s' count:'%d'", h.metricName, len(distributions))
+	hgs := map[senders.HistogramGranularity]bool{h.Granularity: true}
+	for _, distribution := range distributions {
+		sender.SendDistribution(h.metricName, distribution.centroids, hgs, distribution.timestamp.Unix(), h.source, h.tags)
+	}
+}
+
+func (h *Histogram) getDistributions() []distribution {
```

Comment:
rename to `distributions()`. It's idiomatic in go to omit the get prefixes getters: https://golang.org/doc/effective_go.html#Getters